cfme/tests/physical_infrastructure/test_providers.py

```diff
@@ -0,0 +1,48 @@
+# -*- coding: utf-8 -*-
+import uuid
+import fauxfactory
+
+import pytest
+
+from copy import copy, deepcopy
+
+from cfme.utils import error
+from cfme.base.credential import Credential
+from cfme.common.provider_views import (PhysicalProviderAddView,
+                                        PhysicalProvidersView)
+from cfme.physical.provider.lenovo import PhysicalProvider, LenovoProvider, LenovoEndpoint
+from cfme.utils import testgen, version
+from cfme.utils.update import update
+from cfme.utils.blockers import BZ
+from cfme import test_requirements
+from cfme.utils.appliance import get_or_create_current_appliance
+
+
+def test_provider_crud():
```

Comment:
**REQUIRED:** So actually, you should be able to utilize the existing ```test_provider_crud``` test. We also don't hard code the provider, we let ```testgen``` do it for us, so you would make the following changes to make this a legitimate test, excluding imports:

```
pytestmark = pytest.mark.provider([LenovoProvider], scope='module')

def test_provider_crud(provider):
    provider.create()
    provider.validate_stats(ui=True)
...
```

You can then invoke the test run by doing

```miq-runtest cfme/tests/physical_infrastructure/test_providers.py -k test_provider_crud --use-provider lenovo```